spacer/torch_utils.py

```diff
@@ -33,6 +34,10 @@ def load_weights(model: Any,
     :param modelweighs_path: pretrained model weight from new CoralNet
     :return: well trained model
     """
+    with open(modelweighs_path, 'rb') as fp:
+        sha256 = hashlib.sha256(fp.read()).hexdigest()
+    assert sha256 == \
+           'c3dc6d304179c6729c0a0b3d4e60c728bdcf0d82687deeba54af71827467204c'
```

Comment:
`load_weights`  is intended to support any EfficientNet implementation, right? So the sha needs to be specific to this `feature_extractor_name`. I suggest you a dict in the main config file indexed on the `feature_extractor_name`. Can you add the same for the caffe extractor pls.?

Also, it looks like you are loading the weights twice. Once to calculate the sha, and then again below. Can you try to read it just once?

Reply:
sure, i can add the same for the Caffe, do you want me to move the sha to the config?

> Also, it looks like you are loading the weights twice. Once to calculate the sha, and then again below. Can you try to read it just once?

ok, working on it

Reply:
Yeah. Each feature_extractor should have a weight check-sum to make sure we are loading the weights we think we are. These should be stored in the main config file.

Reply:
> `load_weights` is intended to support any EfficientNet implementation, right? So the sha needs to be specific to this feature_extractor_name

Yes, `load_weights` can support any EfficientNet implementation and actually `class EfficientNetExtractor(FeatureExtractor)` support any EfficientNet so I dictionarize the sha indexed on the `model_name` instead of `feature_extractor_name`, what do you think?

Checksum added on the Caffe and modify the checksum in Pytorch to load the weights only once.